includes/HookHandlers/RequestWiki.php

```diff
@@ -2,118 +2,606 @@
 
 namespace Miraheze\MirahezeMagic\HookHandlers;
 
+use MediaWiki\Config\Config;
+use MediaWiki\Config\ServiceOptions;
+use MediaWiki\HTMLForm\Field\HTMLToggleSwitchField;
 use MediaWiki\User\User;
+use Miraheze\CreateWiki\Hooks\CreateWikiAfterCreationWithExtraDataHook;
 use Miraheze\CreateWiki\Hooks\RequestWikiFormDescriptorModifyHook;
 use Miraheze\CreateWiki\Hooks\RequestWikiQueueFormDescriptorModifyHook;
 use Miraheze\CreateWiki\RequestWiki\FormFields\DetailsWithIconField;
 use Miraheze\CreateWiki\RequestWiki\RequestWikiFormUtils;
 use Miraheze\CreateWiki\Services\WikiRequestManager;
+use Miraheze\ManageWiki\Helpers\ManageWikiExtensions;
+use Miraheze\ManageWiki\Helpers\ManageWikiSettings;
 
 class RequestWiki implements
+	CreateWikiAfterCreationWithExtraDataHook,
 	RequestWikiFormDescriptorModifyHook,
 	RequestWikiQueueFormDescriptorModifyHook
 {
 
+	private ServiceOptions $options;
+
+	public function __construct( ServiceOptions $options ) {
+		$this->options = $options;
+	}
+
+	public static function factory( Config $mainConfig ): self {
```

Comment:
You probably already tested and checked this, but why is this necessary?

Reply:
Because it is common convention to use ServiceOptions here for injection.